extensions/commands/art/cmd_promote.py

```diff
@@ -0,0 +1,122 @@
+import json
+import os.path
+
+from conan.api.conan_api import ConanAPI
+from conan.api.output import ConanOutput
+from conan.cli.command import conan_command
+from conans.model.recipe_ref import RecipeReference
+from conans.model.package_ref import PkgReference
+from conan.api.model import MultiPackagesList
+from conan.errors import ConanException
+
+from utils import api_request, assert_server_or_url_user_password
+from cmd_server import get_url_user_password
+
+
+def _get_export_path_from_rrev(rrev):
+    recipe_ref = RecipeReference.loads(rrev)
+    path = f"_/{recipe_ref.name}/{recipe_ref.version}"
+    if recipe_ref.revision:
+        path += f"/_/{recipe_ref.revision}/export/"
+    return path
+
+
+def _get_path_from_pref(pref):
+    package_ref = PkgReference.loads(pref)
+    recipe_ref = package_ref.ref
+
+    path = f"_/{recipe_ref.name}/{recipe_ref.version}/_/{recipe_ref.revision}/package/{package_ref.package_id}"
+    if package_ref.revision:
+        path += f"/{package_ref.revision}"
+    return path
+
+
+def _request(url, user, password, request_type, request_url):
+    try:
+        return json.loads(api_request(request_type, f"{url}{request_url}", user, password))
+    except Exception as e:
+        raise ConanException(f"Error requesting {request_url}: {e}")
+
+
+def _promote_path(url, user, password, origin, destination, path):
+    ConanOutput().subtitle(f"Promoting {path}")
+    # The copy api creates a subfolder if the destination already exists, need to check beforehand to avoid this
+    try:
+        # This first request will raise a 404 if no file is found
+        _request(url, user, password, "get", f"api/storage/{destination}/{path}")
+        ConanOutput().warning("Destination already exists, skipping")
+    except ConanException:
+        _request(url, user, password, "post", f"api/copy/{origin}/{path}?to=/{destination}/{path}&suppressLayouts=0")
+        ConanOutput().success("Promoted file")
+
+
+@conan_command(group="Artifactory")
+def promote(conan_api: ConanAPI, parser, *args):
+    """
+    Promote a pkglist file from an origin Artifactory repository to a destination repository, without downloading the packages locally
```

Comment:
This can also be used to promote just the recipes if the pkglist contains no packages, but it's true that something a bit more clear is needed. Any suggestions that might let the user know that it will promote everything?

Reply:
```suggestion
    Promote Conan recipes and packages in a pkglist file from an origin Artifactory repository to a destination repository, without downloading the packages locally
```
Maybe like so?